tox.ini

```diff
@@ -54,11 +54,43 @@ commands =
     pip-audit {env:EXCLUDES:}
 
 [testenv:docs,docs-py{39,310,311,312}]
-description = Build documentation
+description = Build documentation by building the examples, checking the links and
+    performing quickbuild
 extras = docs
 passenv = BAYBE_DOCS_LINKCHECK_IGNORE
 setenv =
     SMOKE_TEST = true
 commands = 
+    # TODO Can we directly call the corresponding tox commands here?
     python --version
-    python docs/scripts/convert_code_to_documentation.py {posargs}
\ No newline at end of file
+    python docs/scripts/build_examples.py {posargs}
+    python docs/scripts/check_links.py {posargs}
+    python docs/scripts/build_documentation.py {posargs}
+
+
+[testenv:docs-examples,docs-examples-py{39,310,311,312}]
+description = Build examples for documentation
+extras = docs
+setenv =
+    SMOKE_TEST = true
+commands = 
+    python --version
+    python docs/scripts/build_examples.py {posargs}
+
+[testenv:docs-linkcheck,docs-linkcheck-py{39,310,311,312}]
+description = Check links for the documentation
+extras = docs
+setenv =
+    SMOKE_TEST = true
+commands = 
+    python --version
+    python docs/scripts/check_links.py {posargs}
+
+[testenv:docs-quickbuild,docs-quickbuild-py{39,310,311,312}]
+description = Quickly re-build the documentation if it was already built
```

Comment:
how much quicker is this now?

Reply:
On my machine, I get the following times (using force build):
- Full building of documentation including examples (which needs to be done at least once since sphinx needs the files): 342 seconds
- Quickbuilding the documentation afterwards without any changes: 15 seconds
- Quickbuilding the documentation with a minor change (adding a dummy exception): 21 seconds